src/ResourceManager/Profile/Commands.Profile/Default/SetAzureRMDefault.cs

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Profile.Models;
+using Microsoft.Azure.Commands.Profile.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.Azure.Management.Internal.Resources.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Profile.Default
+{
+    /// <summary>
+    /// Cmdlet to set default options. 
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmDefault", DefaultParameterSetName = ResourceGroupNameParameterSet,
+        SupportsShouldProcess = true)]
+    [OutputType(typeof(PSResourceGroup))]
+    public class SetAzureRMDefaultCommand : AzureRMCmdlet
```

Comment:
Same comment, this needs to extend ContextModificationCmdlet, and changes to the context should be inside a ModifyContext call, see cmdlets liek Set-AzureRmContext for the pattern.